src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerServiceTest.java

```diff
@@ -17,60 +17,54 @@
 import org.springframework.kafka.core.KafkaTemplate;
 import org.springframework.kafka.support.SendResult;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.util.ReflectionTestUtils;
 import org.springframework.util.concurrent.ListenableFuture;
 import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
-import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
 import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
 
 @ExtendWith(SpringExtension.class)
-class KafkaProducerTimeEntryTest {
+class KafkaProducerServiceTest {
 
   @Mock
-  private KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+  private KafkaTemplate<String, KafkaEventMessage<TimeEntry>> kafkaTimeEntryTemplate;
```

Comment:
Even though we're using `TimeEntry` as the type constraint for testing, `KafkaProducerService` is now a generic class and I think the naming needs to reflect that, otherwise it misleading what's expected of `KafkaProducerService`.
It might be better to name this just kafkaTemplate

Reply:
Updated